server/routes/api/import_detection/detection.py

```diff
@@ -194,3 +202,148 @@ def detect_columns(col_order: List[Column],
+                        column_idx: 1,
+                    },
+                    ...
+                ],
+            }
+        ]
+    """
+    # Parse and validate the query params.
+    column_order: List[Column] = []
+    js = json.loads(request.get_json())
+    try:
+        column_idx_order: List[int] = [int(v) for v in js["column_idx_order"]]
+        column_ids: Dict[int, str] = {
+            int(k): v for k, v in js["column_ids"].items()
+        }
+        column_headers: Dict[int, str] = {
+            int(k): v for k, v in js["column_headers"].items()
+        }
+        column_values: Dict[int, List[str]] = {
+            int(k): v for k, v in js["column_values"].items()
+        }
+
+        if not column_idx_order or not column_ids or not column_headers or not column_values:
+            raise KeyError()
+
+        if not utils.check_list_instance(column_idx_order, int):
+            raise TypeError(
+                "column_idx_order must be a list of integers (column indices).")
+        if not utils.check_dict_instance(column_ids, int, str):
+            raise TypeError(
+                "column_ids must be a Dict mapping integers (column indices) to strings (column ids)."
+            )
+        if not utils.check_dict_instance(column_headers, int, str):
+            raise TypeError(
+                "column_headers must be a Dict mapping integers (column indices) to strings (column headers)."
+            )
+        if not utils.check_dict_instance(column_values, int, List):
+            raise TypeError(
+                "column_values must be a Dict mapping integers (column indices) to a list of strings (column values)."
+            )
+
+        # Also check if all column values are of type str.
+        for _, vals in column_values.items():
+            for v in vals:
+                if not isinstance(v, str):
+                    raise TypeError(
+                        "All provided column values must be of type string.")
+
+        # Now convert the inputs to the Column type.
+        for col_idx in column_idx_order:
+            if not col_idx in column_ids or not col_idx in column_headers or not col_idx in column_values:
+                raise AttributeError(
+                    "All column indices in column_idx_order must be found as keys in column_ids, column_headers and column_values"
+                )
+            column_order.append(
+                Column(id=column_ids[col_idx],
+                       header=column_headers[col_idx],
+                       column_idx=col_idx))
+
+    except KeyError as key_error:
+        return f"TypeError: Missing request parameter {key_error.args[0]}", BAD_REQUEST_CODE
+
+    except TypeError as type_error:
+        return f"TypeError: Incorrect request parameter type: {type_error.args[0]}", BAD_REQUEST_CODE
+
+    except AttributeError as attrb_error:
+        return f"AttributeError: {attrb_error.args[0]}", BAD_REQUEST_CODE
+
+    if not column_order:
```

Comment:
If we get here, that would just mean there were no valid columns right? and not because there were any actual problems with the request (besides things that have already been caught)? So I don't think this should be a bad request, maybe just empty result with success code?

Reply:
Yeah, fair enough.